src/sources/StargateLPYieldSource.sol

```diff
@@ -0,0 +1,152 @@
+    IERC20 public immutable override yieldToken;
+
+    /// @notice LP staking contract
+    ILPStaking public immutable staking;
+
+    /// @notice Uniswap router
+    IUniswapV2Router02 public router;
+
+    /// @notice Pool Id
+    uint256 public pid;
+
+    /**
+     * @notice Contract constructor
+     * @param _pid Staking pool id
+     * @param _lpToken LP token
+     * @param _staking Staking contract
+     * @param _router Router for swap
+     */
+    constructor(
+        uint256 _pid,
+        IERC20 _lpToken,
+        ILPStaking _staking,
+        IUniswapV2Router02 _router
+    ) {
+        require(address(_lpToken) != address(0), "LP: zero address");
+        require(address(_staking) != address(0), "Staking: zero address");
+
+        pid = _pid;
+        staking = _staking;
+        sourceToken = _lpToken;
+        router = _router;
+        yieldToken = IERC20(staking.stargate());
+    }
+
+    /**
+     * @notice Returns pending STG yield
+     */
+    function pendingYield() external view override returns (uint256) {
+        uint256 balance = yieldToken.balanceOf(address(this));
+        return balance + staking.pendingStargate(pid, address(this));
+    }
+
+    /**
+     * @notice Total deposited lp token
+     */
+    function totalDeposit() external view override returns (uint256) {
+        return _totalDeposit();
+    }
+
+    /**
+     * @notice Stake lp tokens
+     */
+    function deposit(uint256 amount) external override onlyOwner {
+        sourceToken.safeTransferFrom(msg.sender, address(this), amount);
+        _deposit(amount);
+    }
+
+    /**
+     * @notice Withdraw lp tokens
+     * @dev Harvest happens automatically by stargate
+     */
+    function withdraw(
+        uint256 amount,
+        bool claim,
+        address to
+    ) external override onlyOwner {
+        staking.withdraw(pid, amount);
+        uint256 balance = sourceToken.balanceOf(address(this));
+        if (amount > balance) {
+            amount = balance;
+        }
+        sourceToken.safeTransfer(to, amount);
+        if (!claim) {
+            _deposit(amount);
+        }
+    }
+
+    /**
+     * @notice Harvest rewards
+     */
+    function harvestAndConvert(
+        IERC20 outToken,
+        uint256 amount
+    ) external override onlyOwner returns (uint256 yieldAmount, uint256 actualOut) {
+        // harvest by withdraw
+        ILPStaking.UserInfo memory info = staking.userInfo(pid, address(this));
+        staking.withdraw(pid, info.amount);
+
+        // redeposit asset
+        _deposit(info.amount);
+
+        if (amount > 0) {
+            // swap yield into outToken
+            yieldToken.safeApprove(address(router), amount);
+            address[] memory path = new address[](2);
+            path[0] = address(yieldToken);
+            path[1] = address(outToken);
+            uint256[] memory amounts = router.swapExactTokensForTokens(amount, 0, path, msg.sender, block.timestamp);
```

Comment:
@icyage what are we swapping here? There actually does not need to be a swap involved as the goal is to be able to deposit the native yield into earthquake.